scripts/dts/extract/clocks.py

```diff
@@ -42,27 +181,45 @@ def _extract_consumer(self, node_address, yaml, clocks, names, defs, def_label):
                                     str(clock_provider)))
                 clock_provider_node_address = phandles[cell]
                 clock_provider = reduced[clock_provider_node_address]
-                clock_provider_compat = get_compat(clock_provider_node_address)
-                clock_provider_bindings = yaml[clock_provider_compat]
-                clock_provider_label = self.get_node_label_string( \
-                                                clock_provider_node_address)
-                nr_clock_cells = int(clock_provider['props'].get(
-                                     '#clock-cells', 0))
-                clock_cells_string = clock_provider_bindings.get(
-                    'cell_string', 'CLOCK')
-                clock_cells_names = clock_provider_bindings.get(
-                    '#cells', ['ID', 'CELL1',  "CELL2", "CELL3"])
+                clock_nr_cells = int(clock_provider['props'].get('#clock-cells', 0))
                 clock_cells = []
             else:
                 clock_cells.append(cell)
             clock_cell_index += 1
-            if clock_cell_index > nr_clock_cells:
-                # clock consumer device - clocks info
-                #####################################
+            if clock_cell_index > clock_nr_cells:
+
+                 # generate EDTS
+                edts_insert_device_property(clock_consumer_device_id,
+                    'clocks/{}/name'.format(clock_index),
+                    self._clock_input_name(clock_consumer, clock_index))
+                edts_insert_device_property(clock_consumer_device_id,
+                    'clocks/{}/provider'.format(clock_index),
+                    edts_device_id(clock_provider_node_address))
+                for prop in ('clock-ranges',):
```

Comment:
What if 'clock-ranges' is not present?
More generally, I don't think so much properties should be hardcoded in these files.
Once again difficult to review when no example of usage is available.
I'd prefer clock is kept to existing area for now.
Then we can add functionalities to clock subsystem with evolution in dt and in drivers, but this should be done in a separate PR.

Reply:
> What if 'clock-ranges' is not present?

Property will not be filled into EDTS.

> More generally, I don't think so much properties should be hardcoded in these files

There is no hard coding. See the possible properties listed in the header. If they are i n the DTS, they will be inserted into the EDTS.

> Once again difficult to review when no example of usage is available.
> I'd prefer clock is kept to existing area for now.

Ok. I understand. I have a driver based on these properties (like I have a driver based on the pinctrl properties). The properties are a prerequisite for clock my driver to work.